client/src/ui/deckbuilder/DeckBuilderColumn.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <div
+      class="column"
+      @dragover="dragOver"
+      @dragleave="dragEnd"
+      @dragend="dragEnd"
+      @drop="drop"
+  >
+    <div
+        v-for="(card, index) in column"
+        class="card"
+    >
+      <img
+          class="card-img"
+          :src="getImageSrc(card)"
+          :alt="card.name"
+          :data-index="index"
+          @dragstart="dragStart"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import {MtgCard} from "../../draft/DraftState.js";
+  import {CardColumn, CardMove} from "../../state/store.js";
+
+  export default Vue.extend({
+    name: 'DeckBuilderColumn',
+
+    props: {
+      column: {
+        type: Array as () => CardColumn
+      },
+      columnIndex: {
+        type: Number
+      },
+      deckIndex: {
+        type: Number
+      },
+      maindeck: {
+        type: Boolean
+      }
+    },
+
+    methods: {
+      getImageSrc(card: MtgCard): string {
+        if (process.env.NODE_ENV == 'development') {
+          return `http://api.scryfall.com/cards/${card.set}/`
+              + `${card.collector_number}?format=image&version=normal`;
+        } else {
+          return `/proxy/${card.set}/${card.collector_number}`;
+        }
+      },
+
+      dragStart(e: DragEvent) {
+        if (e.dataTransfer) {
+          const targetElement = <HTMLElement>e.target;
+          const cardMove: CardMove = {
+            deckIndex: this.deckIndex,
+            sourceMaindeck: this.maindeck,
+            sourceColumnIndex: this.columnIndex,
+            sourceCardIndex: Number(targetElement.dataset["index"]),
+            targetMaindeck: false,
+            targetColumnIndex: 0,
+          };
+          e.dataTransfer.setData("text/plain", JSON.stringify(cardMove));
+          e.dataTransfer.effectAllowed = "move";
+        }
+      },
+
+      dragOver(e: DragEvent) {
+        e.preventDefault();
+        if (e.dataTransfer) {
+          e.dataTransfer.dropEffect = "move";
+          this.$el.classList.add("columnDrop");
+        }
+      },
+
+      dragEnd(e: DragEvent) {
+        e.preventDefault();
+        if (e.dataTransfer) {
+          this.$el.classList.remove("columnDrop");
```

Comment:
These class modifications will get nuked as soon as the component re-renders. The idiomatic way to do this in Vue is to
(a) Add something like a `isDragging` field to the component's `data` (see https://vuejs.org/v2/guide/components.html)
(b) In the HTML template, conditionally add the class columnDrop based on `isDragging` (see https://vuejs.org/v2/guide/class-and-style.html)

Reply:
Since these classes are only active during the drag, I don't think we care about a re-render (and I'm not sure if triggering a re-render would screw up the dragging).